src/preprocessing/preprocessing.py

```diff
@@ -0,0 +1,190 @@
+file_path = os.path.join(current_dir, "../data/last_snapshot.csv")
+
+sys.path.append(parent_dir)
+from logger import get_logger
+
+log = get_logger()
+
+
+class Preprocessing:
+    def __init__(self, df, filter_null_data=True):
+        self.preprocessed_df = df.copy()
+
+        self.filter_bool = filter_null_data
+        # columns that would be added later after one-hot encoding each class
+        self.added_classes = []
+        self.numerical_data = [
+            "google_places_rating",
+            "google_places_user_ratings_total",
+            "google_places_confidence",
+            "reviews_sentiment_score",
+            "review_avg_grammatical_score",
+            "review_polarization_score",
+            "review_highest_rating_ratio",
+            "review_lowest_rating_ratio",
+            "review_rating_trend",
+        ]
+        # numerical data that need scaling
+        self.data_to_scale = ["google_places_user_ratings_total"]
+
+        # categorical data that needs one-hot encoding
+        self.categorical_data = [
+            "number_country",
+            "number_area",
+            "google_places_detailed_type",
+            "review_polarization_type",
+        ]
+
+    def filter_out_null_data(self):
+        self.preprocessed_df = self.preprocessed_df[
+            self.preprocessed_df["google_places_confidence"].notnull()
+        ]
+
+    def fill_missing_values(self, column, strategy="constant"):
+        imputer = SimpleImputer(strategy=strategy)
+        self.preprocessed_df[column] = imputer.fit_transform(
+            self.preprocessed_df[[column]]
+        )
+        return self.preprocessed_df
+
+    def standard_scaling(self, column):
+        # scales the data in such that the mean of the data becomes 0 and the standard deviation becomes 1.
+        scaler = StandardScaler()
+        self.preprocessed_df[column] = scaler.fit_transform(
+            self.preprocessed_df[[column]]
+        )
+        return self.preprocessed_df
+
+    def min_max_scaling(self, column):
+        # scales the data to a given range, usually between 0 and 1.
+        scaler = MinMaxScaler()
+        self.preprocessed_df[column] = scaler.fit_transform(
+            self.preprocessed_df[[column]]
+        )
+        return self.preprocessed_df
+
+    def robust_scaling(self, column):
+        scaler = RobustScaler()
+        self.preprocessed_df[column] = scaler.fit_transform(
+            self.preprocessed_df[[column]]
+        )
+        return self.preprocessed_df
+
+    def normalization(self, column):
+        scaler = Normalizer()
+        self.preprocessed_df[column] = scaler.fit_transform(
+            self.preprocessed_df[[column]]
+        )
+        return self.preprocessed_df
+
+    def remove_outliers_zscore(self, column):
+        z_scores = stats.zscore(self.preprocessed_df[[column]])
+        self.preprocessed_df[column] = self.preprocessed_df[
+            (z_scores < 3) & (z_scores > -3)

```

Comment:
might use top-level variables here 
```suggestion
            (z_scores < THRESHOLD) & (z_scores > -THRESHOLD)
```

Reply:
Addressed.